transcom/mymove
# [MB-7422] Send all moves to service counseling

## Description

Send all moves to service counseling by setting their status to the new `MoveStatusNeedsServiceCounseling` status (from: https://github.com/transcom/mymove/pull/6322). We added a service object to handle this logic (anticipating that later we will need to determine which moves get this new status and which get the old "submitted" status), and put this behind a feature flag called `FEATURE_FLAG_SERVICE_COUNSELING`.

## Reviewer Notes

Because this is adding a feature flag and will (likely) affect other teams, is there some place I should be documenting this?

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make db_dev_e2e_populate
make server_run
make client_run
```

## Code Review Verification Steps

1. This work is behind a feature flag called `FEATURE_FLAG_SERVICE_COUNSELING`. It is set to `true` by default in `.envrc`. When the flag is enabled, any new move that you submit will still show up in the TOO queue, but it should have the status `NEEDS SERVICE COUNSELING` and you won't be able to approve it because approval requires completion of Service Counseling, which isn't implemented yet. 
2. Run the MyMove app, and login as `hhg@only.unsubmitted` test user
3. Go to Review and Submit, click Next
4. Enter a signature, open the network tab, and click on Complete
5. The submit POST should respond with an object that includes `status: "NEEDS SERVICE COUNSELING"`

To verify that moves are not sent to service counseling when the flag is off:
1. Change the feature flag in `.envrc.local` to `export FEATURE_FLAG_SERVICE_COUNSELING=false`, run `direnv allow` after you make the change.
2. Run `make db_dev_e2e_populate` to reset the test data, then run the server/client
3. In the MyMove app, login to `hhg@only.unsubmitted` test user
4. Go to Review and Submit, click Next
5. Enter a signature, open the network tab, and click on Complete
6. The submit POST should respond with an object that includes `status: "SUBMITTED"`


* Any new migrations/schema changes:

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-7422) for this change
